src/hooks/useEntitlement.ts

```diff
@@ -35,9 +34,9 @@ const notifyOnChangeProps = ['data' as const, 'isLoading' as const];
  *
  *  */
 const useEntitlement: UseEntitlement = (playlistItem) => {
-  const checkoutService: CheckoutService = useService(({ checkoutService }) => checkoutService);
+  let checkoutController: CheckoutController | null = null;
```

Comment:
I think it's simpler to handle the user id checking elsewhere instead of tying the module resolution to it.
```suggestion
  const checkoutController = getModule(CheckoutController);
```

Reply:
`useEntitlement` is used for the `StartWatchingButton` which exists for both AVOD and SVOD integrations. When using AVOD `CheckoutController` is not defined.

Reply:
I think it's fine if it's undefined, but I think I see the problem... the getModule method doesn't allow for optional resolution. Can we just add a flag argument `optional: boolean = false`?

Reply:
Do you suggest to wrap the controller invocation in try / catch and return `undefined` in case `required` is not specified?

Reply:
I would suggest to update the getModule code to something like this:
```
export const getModule = <T>(type: interfaces.Newable<T>, args: {required: boolean} = {required: true}): T | undefined => {
  const all = container.getAll(type);

  // This app expects 0 or 1 instances of each type and no more
  if (all.length > 1) {
    throw new Error(`Type not unique for ${type}`);
  }

  if (all.length === 1) {
    return all[0];
  }

  if (!container.isBound(type) && !args.required) {
    return undefined;
  }

  return container.resolve(type);
};
```

Reply:
The code where you're calling the controller you've already allowed it to be null, so undefined should work just the same. I think this approach will let you clean up a bunch of ternaries with getModule assignments.